### PR TITLE
[Fix](inverted index) fix wrong opt for count_on_index

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

The pushdownCountOnIndex optimization rule in AggregateStrategies was incorrectly applied to COUNT functions with complex child expressions, such as COUNT(CASE WHEN ...). This led to incorrect query results.

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java [hunk 1]
```diff
@@ -136,7 +137,8 @@ public List<Rule> buildRules() {
                     .when(agg -> agg.getGroupByExpressions().isEmpty())
                     .when(agg -> {
                         Set<AggregateFunction> funcs = agg.getAggregateFunctions();
-                        return !funcs.isEmpty() && funcs.stream().allMatch(f -> f instanceof Count && !f.isDistinct());
+                        return !funcs.isEmpty() && funcs.stream().allMatch(f -> f instanceof Count && !f.isDistinct()
+                            && (((Count) f).isCountStar() || f.child(0) instanceof Slot));
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:141]: f.child(0) could be constant (not only literal). f.child(0).isConstant()
example
select count(1) from T;

Comment (airborne12) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:141]:
```
public boolean isCountStar() {
        return isStar
                || children.isEmpty()
                || (children.size() == 1 && child(0) instanceof Literal);
    }
```
isCountStar already included.

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by csun5285 — APPROVED** [review]
LGTM

**Pull request merged**